src/autobot/internal/steps/abdrawserializationstep.cpp

```diff
@@ -18,15 +18,37 @@
 //=============================================================================
 #include "abdrawserializationstep.h"
 
+#include <QFile>
+
 #include "log.h"
 #include "../draw/abpaintprovider.h"
 
 using namespace mu::autobot;
 
 void AbDrawSerializationStep::doRun(AbContext ctx)
 {
-    std::string data;
-    AbPaintProvider::instance()->serialize(data);
-    LOGI() << data;
+    io::path drawDataPath = configuration()->drawDataPath();
+    if (!fileSystem()->exists(drawDataPath)) {
+        Ret ret = fileSystem()->makePath(drawDataPath);
+        if (!ret) {
+            LOGE() << "failed make path: " << drawDataPath;
+            doFinish(ctx);
+            return;
+        }
+    }
+
+    QByteArray data = AbPaintProvider::instance()->serialize();
+
+    io::path scorePath = ctx.val<io::path>(AbContext::Key::ScoreFile);
+    io::path filePath = drawDataPath + "/" + io::basename(scorePath) + ".json";
```

Comment:
I think it is better to move this code to the configuration

Reply:
I don't think so, this is the path for the current score data, depends on the current context